src/Kiota.Builder/Refiners/TypeScriptRefiner.cs

```diff
@@ -1103,9 +1103,30 @@ protected static void AddEnumObject(CodeElement currentElement)
         }
         CrawlTree(currentElement, AddEnumObject);
     }
+    protected static IEnumerable<CodeEnum> selectEnums(IEnumerable<CodeProperty> props)
+    {
+        return props.Select(static x => x.Type).OfType<CodeType>().Select(static x => x.TypeDefinition).OfType<CodeEnum>();
+    }
 
     protected static void AddEnumObjectUsings(CodeElement currentElement)
     {
+        if (currentElement is CodeProperty codeProperty && codeProperty.Kind is CodePropertyKind.RequestBuilder && codeProperty.Type is CodeType codeType && codeType.TypeDefinition is CodeClass codeClass)
```

Comment:
I'm not 100% sure I'm making the right selection here, and if I'm missing any case, I hope @baywet will tell me 🙏 🙂

Reply:
@andreaTP I'm not sure this is the right thing to do here. By the time this function is called, codeclass instances have been converted to interfaces and methods to functions. 
Also, if two/three/four.... properties exist in a class that match this criterion, the class methods will be processed two/three/four.... times. So we should check if parent has a child that matches and then process all the children.

Probably something like(sorry, haven't tested this though)

```cs
        if (currentElement is CodeInterface { Kind: CodeInterfaceKind.RequestBuilder } codeInterface)
        {
            foreach (var propertyMethod in codeInterface.Methods)
            {
                if (propertyMethod.ReturnType is CodeType { TypeDefinition: CodeEnum codeEnum })
                {
                    codeInterface.AddUsing(new CodeUsing
                    {
                        Name = codeEnum.Name,
                        Declaration = new CodeType
                        {
                            TypeDefinition = codeEnum.CodeEnumObject
                        }
                    });
                }
            }
        }
```